### ISSUE TITLE
KeyNotFound while commiting

### ISSUE THREAD

**Issue by jecisc** [pharo-vcs/iceberg#442]
It happens from time to time that I get the error 'KeyNotFound' while commiting…
I just got it while working on MaterialDesign.

`KeyNotFound: key #'Seaside-Canvas' not found in Dictionary`

**Comment by jecisc** [comment]
I think this is a problem while commiting extension methods. Because I did not commit anything in 'Seaside-Canvas' but I have extensions that extends classes of this package.

**Comment by jecisc** [comment]
Here is the stack:

```
Dictionary>>errorKeyNotFound:
[ self errorKeyNotFound: key ] in Dictionary>>at: in Block: [ self errorKeyNotFound: key ]
Dictionary>>at:ifAbsent:
Dictionary>>at:
IceRepository>>packageNamed:
[ :packageName :changes | 
| workingCopy snapshot patcher |
workingCopy := (RPackageOrganizer default packageNamed: packageName)
	mcWorkingCopy.
snapshot := (self packageNamed: packageName) loadedVersion snapshot.
patcher := MCPatcher snapshot: snapshot.
pickedCollection
	select: [ :each | each operation notNil ]
	thenDo: [ :each | each operation applyTo: patcher ].
versions
	add:
		(MCVersion new
			setPackage: workingCopy package
			info: (IceMCVersionInfo package: workingCopy package message: message)
			snapshot: patcher patchedSnapshot
			dependencies: #()) ] in IceRepository>>commitCherryPick:withMessage:andParents: in Block: [ :packageName :changes | ...
SmallDictionary>>keysAndValuesDo:
IceRepository>>commitCherryPick:withMessage:andParents:
IceRepository>>commitCherryPick:withMessage:
IceCherryPickCommitModel>>basicCommitWithMessage:
[ self basicCommitWithMessage: text asString ] in [ [ self basicCommitWithMessage: text asString ]
	on: IceGitUsernameOrEmailNotFound
	do: [ :e | 
		self configGitUsernameAndEmailOnCancel: cancelBlock.
		e retry ].
self resetText ] in IceCherryPickCommitModel(IceCommitModel)>>basicCommitWithMessage:ifCancel: in Block: [ self basicCommitWithMessage: text asString ]
BlockClosure>>on:do:
[ [ self basicCommitWithMessage: text asString ]
	on: IceGitUsernameOrEmailNotFound
	do: [ :e | 
		self configGitUsernameAndEmailOnCancel: cancelBlock.
		e retry ].
self resetText ] in IceCherryPickCommitModel(IceCommitModel)>>basicCommitWithMessage:ifCancel: in Block: [ [ self basicCommitWithMessage: text asString ]...
BlockClosure>>on:do:
IceCherryPickCommitModel(IceCommitModel)>>basicCommitWithMessage:ifCancel:
[ :bar | 
bar label: ('Commiting changes to {1}' format: {self repository}).
self basicCommitWithMessage: text ifCancel: [ ^ self ].
pushBoolean
	ifTrue: [ | remote |
		remote := self repository pushRemote.
		bar label: ('Pushing changes to {1}' format: {remote}).
		self pushRepository: self repository andThen: [  ].
		UIManager default
			inform: ('Uploaded changes to {1}' format: {remote}) ] ] in IceCherryPickCommitModel(IceCommitModel)>>commitPushing: in Block: [ :bar | ...
[ :bar | aBlock value: bar ] in MorphicUIManager>>informUserDuring: in Block: [ :bar | aBlock value: bar ]
BlockClosure>>cull:
[ ^ block cull: self ] in [ self prepareForRunning.
CurrentJob value: self during: [ ^ block cull: self ] ] in Job>>run in Block: [ ^ block cull: self ]
[ activeProcess psValueAt: index put: anObject.
aBlock value ] in CurrentJob(DynamicVariable)>>value:during: in Block: [ activeProcess psValueAt: index put: anObject....
BlockClosure>>ensure:
CurrentJob(DynamicVariable)>>value:during:
CurrentJob class(DynamicVariable class)>>value:during:
[ self prepareForRunning.
CurrentJob value: self during: [ ^ block cull: self ] ] in Job>>run in Block: [ self prepareForRunning....
BlockClosure>>ensure:
Job>>run
MorphicUIManager(UIManager)>>displayProgress:from:to:during:
MorphicUIManager>>informUserDuring:
IceCherryPickCommitModel(IceCommitModel)>>commitPushing:
[ self commitPushing: true ] in IceCherryPickCommitModel(IceCommitModel)>>initializePresenter in Block: [ self commitPushing: true ]

```

**Comment by jecisc** [comment]
Current workaround: I removed the methods from the extension protocol, then I recategorized them in the extension protocol and I could commit after that.

**Comment by theseion** [comment]
This should be fixed by PR #452 (see issue #450).

**Comment by jecisc** [comment]
Since I don't have a reproducable case I'll just close the issue for now and if I see that again I'll reopen.
